### PR TITLE
[Merged by Bors] - feat: symmetric monoidal structure on graded objects

### PR DESCRIPTION
**Pull request by joelriou**
In this PR, we construct the braiding for the monoidal structure on graded objects (#14457). We show it is symmetric if the original category is symmetric.

---

- [x] depends on: #14457
- [x] depends on: #14343
- [x] depends on: #7425
- [x] depends on: #8237 
- [x] depends on: #8239
- [x] depends on: #8242
- [x] depends on: #10667
- [x] depends on: #10700 
- [x] depends on: #10701
- [x] depends on: #10747
- [x] depends on: #11676
- [x] depends on: #11687
- [x] depends on: #11701
- [x] depends on: #11703
- [x] depends on: #14339

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/GradedObject/Monoidal.lean [hunk 1]
```diff
@@ -271,10 +296,351 @@ lemma associator_naturality (f₁ : X₁ ⟶ Y₁) (f₂ : X₂ ⟶ Y₂) (f₃
...
+lemma ι_descTensor (j : I) (f : ∀ (i₁ i₂ : I) (_ : i₁ + i₂ = j), X₁ i₁ ⊗ X₂ i₂ ⟶ A)
+    (i₁ i₂ : I) (hi : i₁ + i₂ = j) :
+    ιTensorObj X₁ X₂ i₁ i₂ j hi ≫ descTensor f = f i₁ i₂ hi := by
+  apply Cofan.IsColimit.fac
+
+@[ext]
+lemma tensorObj_ext {j : I} (f g : (X₁ ⊗ X₂) j ⟶ A)
+    (h : ∀ (i₁ i₂ : I) (hi : i₁ + i₂ = j),
+      ιTensorObj X₁ X₂ i₁ i₂ j hi ≫ f = ιTensorObj X₁ X₂ i₁ i₂ j hi ≫ g) : f = g :=
+  Monoidal.tensorObj_ext f g h-/
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/GradedObject/Monoidal.lean:618]: Do you intend to keep this in the PR? Should it be un-commented?

**Review by jcommelin — COMMENTED** [review]
There are some docs missing. But this looks pretty stable already.

File: Mathlib/CategoryTheory/GradedObject/Monoidal.lean (hunk 1 above)
Comment (joelriou) [reply Mathlib/CategoryTheory/GradedObject/Monoidal.lean]: I have removed this for now. In the future, we may have to develop such an API (which would basically just duplicate the API of the internals of the construction of the monoidal category instance).

**Comment by jcommelin** [comment]
@joelriou Is this still WIP?

**Comment by joelriou** [comment]
> @joelriou Is this still WIP?

Indeed, it is now ready for review.

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**